lib/dispatch-rider/publisher.rb

```diff
@@ -33,11 +35,15 @@ def register_channel(service, name, options = {})
       self
     end
 
-    def publish(opts = {})
-      options = opts.dup
+    # @param [Hash] original_options
+    def publish(original_options = {})
+      options = original_options.dup
       add_message_id(options[:message])
-      service_channel_mapper.map(options.delete(:destinations)).each do |(service, channels)|
-        notification_service_registrar.fetch(service).publish(options.merge(:to => channels))
+
+      callbacks.invoke(:publish, options[:message][:body]) do
```

Comment:
Hmm passing just body here is kinda weak .. we need all of options[:message] I think .. otherwise we don't get the subject.